src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java

```diff
@@ -73,6 +81,42 @@ public boolean isOperationsUser(RestRequest request) {
         }
         return allowed;
     }
+    
+    @SuppressWarnings("rawtypes")
+    public void assertUser(RestRequest request) throws Exception {
+        final String user = getUser(request);
+        final String token = getBearerToken(request);
+        ConfigBuilder builder = new ConfigBuilder().withOauthToken(token);
+        try (DefaultOpenShiftClient osClient = new DefaultOpenShiftClient(builder.build())) {
+            LOGGER.debug("Verifying user {} matches the given token.", user);
+            Request okRequest = new Request.Builder()
+                .addHeader(AUTHORIZATION_HEADER, "Bearer " + token)
+                .url(osClient.getMasterUrl() + "oapi/v1/users/~") 
+                .build();
+            String username = null;
+            try (Response response = osClient.getHttpClient().newCall(okRequest).execute()){
+                if(response.code() != RestStatus.OK.getStatus()) {
+                    throw new ElasticsearchSecurityException("The user is unauthorized", RestStatus.UNAUTHORIZED);
+                }
+                String body = response.body().string();
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Response: {}", body);
+                }
+                Map<String, Object> userResponse = XContentHelper.convertToMap(new BytesArray(body), false).v2();
+                if(userResponse.containsKey("metadata") && ((Map)userResponse.get("metadata")).containsKey("name")) {
+                    username = (String) ((Map)userResponse.get("metadata")).get("name");
+                }
+            }catch (Exception e) {
+                LOGGER.debug("Exception trying to assertUser '{}'", e, user);
+                throw e;
+            }
+            if(!user.equals(username)) {
+                String message = String.format("The username '%s' does not own the token provided with the request.", username);
+                LOGGER.debug(message);
+                throw new ElasticsearchSecurityException(message, RestStatus.FORBIDDEN);
```

Comment:
does this message appear anywhere for the user?  If so it may be an info leak.  @simo5, thoughts?

Reply:
It may appear in the response to the call, but if we are not supplying the token in the response i that an issue? Also, a proxy or someone manually curling would have this info already?  Is that an issue?

Reply:
generally unauthorized is simply unauthorized without anymore info

Reply:
fixed and changed to unauthorized which is supposed to provide an auth url per the spec.  Im going to forgo adding that as: 1. not sure what to set it to, 2. its not a true web service and you should know where to get your token.